packages/entities/entities-plugins/src/components/free-form/Datakit/flow-editor/modal/EditorContent.vue

```diff
@@ -17,45 +17,56 @@
     </aside>
     <div class="main">
       <EditorMain
-        @click:backdrop="handleBackdropClick"
-        @click:node="handleNodeClick"
+        @click:backdrop="closeProperties"
+        @click:node="selectNodeAndOpenProperties"
       />
     </div>
 
     <NodePropertiesPanel
-      :node="selectedNode"
       :visible="propertiesPanelVisible"
-      @close="propertiesPanelVisible = false"
+      @close="closeProperties"
     />
   </div>
 </template>
 
 <script setup lang="ts">
 import { createI18n } from '@kong-ui-public/i18n'
 import english from '../../../../../locales/en.json'
-import { usePreferences } from '../../composables'
+import { useEditorStore, usePreferences } from '../../composables'
 import EditorMain from './EditorMain.vue'
 import NodePanel from '../node/NodePanel.vue'
 import NodePropertiesPanel from '../node/NodePropertiesPanel.vue'
-import { ref } from 'vue'
+import { ref, watch } from 'vue'
 import type { NodeInstance } from '../../types'
+import { useDebounce } from '@kong-ui-public/core'
 
 const { t } = createI18n<typeof english>('en-us', english)
 
 const { sidePanelExpanded } = usePreferences()
+const { selectNode, selectedNode } = useEditorStore()!
```

Comment:
Should we add a check to this store usage as well, in case we forgot to call `provide` on the parent-level?